OptimoveSDK/optimove-sdk/src/main/java/com/optimove/sdk/optimove_sdk/kumulos/Kumulos.java

```diff
@@ -0,0 +1,521 @@
+
+package com.optimove.sdk.optimove_sdk.kumulos;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.location.Location;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.text.TextUtils;
+import android.util.Base64;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Collections;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import okhttp3.ConnectionSpec;
+import okhttp3.OkHttpClient;
+import com.optimove.sdk.optimove_sdk.BuildConfig;
+
+/**
+ * The Kumulos class is the main public API for calling Kumulos RPC methods and handling push registration
+ */
+public final class Kumulos {
```

Comment:
the reason for having all Kumulos files in 1 package was to keep access package for files that shouldnt be 'public'. Is it possible to somehow restrict access to 'Kumulos' and other internal classes?

Reply:
Im not sure consumers need access to the Kumulos / OptiMobile static anymore, can we change this to internal or protected level?

Reply:
in android it's mb not super easy as classes using it are in a different package.